test/recipes/20-test_mac.t

```diff
@@ -0,0 +1,133 @@
+#! /usr/bin/env perl
+# Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+
+use strict;
+use warnings;
+
+use OpenSSL::Test;
+use OpenSSL::Test::Utils;
+use Storable qw(dclone);
+
+setup("test_mac");
+
+my @mac_tests = (
+    { cmd => [qw{openssl mac -name SipHash -ctrl hexkey:000102030405060708090A0B0C0D0E0F}],
+      input => '00',
+      expected => 'da87c1d86b99af44347659119b22fc45',
+      desc => 'SipHash No input' },
+    { cmd => [qw{openssl mac -name HMAC -ctrl digest:SHA1 -ctrl hexkey:000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F202122232425262728292A2B2C2D2E2F303132333435363738393A3B3C3D3E3F}],
+      input => unpack("H*", "Sample message for keylen=blocklen"),
+      expected => '5FD596EE78D5553C8FF4E72D266DFD192366DA29',
+      desc => 'HMAC SHA1' },
+    { cmd => [qw{openssl mac -name CMAC -ctrl cipher:AES-256-CBC -ctrl hexkey:0B122AC8F34ED1FE082A3625D157561454167AC145A10BBF77C6A70596D574F1}],
+      input => '498B53FDEC87EDCBF07097DCCDE93A084BAD7501A224E388DF349CE18959FE8485F8AD1537F0D896EA73BEDC7214713F',
+      expected => 'F62C46329B41085625669BAF51DEA66A',
+      desc => 'CMAC AES-256-CBC' },
+    { cmd => [qw{openssl mac -name Poly1305 -ctrl hexkey:02000000000000000000000000000000ffffffffffffffffffffffffffffffff}],
+      input => '02000000000000000000000000000000',
+      expected => '03000000000000000000000000000000',
+      desc => 'Poly1305 (wrap 2^128)' },
+    { cmd => [qw{openssl mac -name GMAC -ctrl cipher:AES-256-GCM -ctrl hexkey:4C973DBC7364621674F8B5B89E5C15511FCED9216490FB1C1A2CAA0FFE0407E5 -ctrl hexiv:7AE8E2CA4EC500012E58495C}],
+      input => '68F2E77696CE7AE8E2CA4EC588E541002E58495C08000F101112131415161718191A1B1C1D1E1F202122232425262728292A2B2C2D2E2F303132333435363738393A3B3C3D3E3F404142434445464748494A4B4C4D0007',
+      expected => '00BDA1B7E87608BCBF470F12157F4C07',
+      desc => 'GMAC' },
+    { cmd => [qw{openssl mac -name KMAC128 -ctrl hexkey:404142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F -ctrl xof:0}],
+      input => '00010203',
+      expected => 'E5780B0D3EA6F7D3A429C5706AA43A00FADBD7D49628839E3187243F456EE14E',
+      desc => 'KMAC128' },
+    { cmd => [qw{openssl mac -name KMAC256 -ctrl hexkey:404142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F -ctrl custom:'My Tagged Application'}],
```

Comment:
Unfortunately, `qw` doesn't pay attention to quotes, neither single or double.  It's easily proven with these perl lines:

``` perl
@foo = qw(one 'two three'); foreach (@foo) { print $_,"\n"; }
@foo = qw(one "two three"); foreach (@foo) { print $_,"\n"; }
```

output:

```
one
'two
three'
one
"two
three"
```

This means that any time you're using quoted strings in your command, you must split the list manually, i.e. something like this:

``` perl
cmd => [qw{openssl mac -name KMAC256 -ctrl
           hexkey:404142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F
           -ctrl}, 'custom:My Tagged Application'],
```

Note that under the hood, `run` will quote any single element of the list appropriately for the platform, so you don't need to do that part yourself.

Reply:
This is the reason the test is failing in Appveyor right now.